docs/proposals/proposal-audit-log.md

```diff
@@ -0,0 +1,256 @@
+      INSERT INTO audit_log (user_id, user_email, organization_id, project_id, project_name, target_id, target_name, schema_version_id, event_action, event_details)
+      VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+    `;
+    return clickhouse.query(query, [
+      userId,
+      userEmail,
+      organizationId,
+      projectId,
+      targetId,
+      schemaVersionId,
+      eventKind,
+      eventAction,
+      JSON.stringify(details),
+    ])
+  };
+}
+
+// sample usage - always await unless performance is critical
+await this.auditLog.logAuditEvent({
+  userId: '690ae6ae-30e7-4e6c-8114-97e50e41aee5',
+  organizationId: 'da2dbbf8-6c03-4abf-964d-8a2d949da5cb',
+  action: 'joinOrganization',
+})
+```
+
+We would call this `logAuditEvent` function in all the places in the codebase to log the events
+listed in the next section.
+
+## Audit log events
+
+Left column shows the event action name, right column shows a human readable example of the event message which we would show in the UI. Keep in mind in the UI we always show the actor of the event, so we don't need to include it in the event message.
+
+| Event Action name              | Human Readable Event sample                                                                               |
+|--------------------------------|-----------------------------------------------------------------------------------------------------------|
+| USER_INVITED                   | User **jane@acme.com** was invited to **Acme**.                                                           |
+| USER_JOINED                    | User **john@acme.com** joined **Acme**. Approved by **admin@acme.com**. Referrer: **jane@acme.com**.      |
+| EXPIRED_INVITE_HIT             | User **john@acme.com** tried to join **Acme** with an expired/invalid invite. Referrer: **jane@acme.com** |
+| PROJECT_CREATED                | User **john@acme.com** created a project named **Project Alpha**.                                         |
+| TARGET_CREATED                 | User **john@acme.com** created a target named **Project Alpha**.                                          |
+| ROLE_CREATED                   | Admin **admin@acme.com** created a new role **SAMPLE_ROLE**.                                              |
+| ROLE_ASSIGNED                  | Admin **admin@acme.com** assigned a new role to user **john@acme.com**.                                   |
+| USER_REMOVED                   | Admin **admin@acme.com** removed user **john@acme.com**.                                                  |
+| ORG_TRANSFERRED                | Admin **admin@acme.com** transferred ownership.                                                           |
+| SCHEMA_CHECKED                 | CI made a schema check for **Project Alpha**.                                                             |
+| SCHEMA_PUBLISH                 | User **john@acme.com** published a new schema version for **Project Alpha**.                              |
```

Comment:
```suggestion
| SCHEMA_PUBLISHED                 | User **john@acme.com** published a new schema version for **Project Alpha**.                              |
```

@n1ru4l do you think we should use past tense everywhere? `SCHEMA_PUBLISHED` vs `PUBLISH_SCHEMA`?

Reply:
I renamed this for consistency with all the other ones